File: front-end/common/userForFormService.js
const $ = require('../utils/request_util');

const insertByList = function (insertList) {
    return $.request({
        url: '/user-for-form/addList',
        data: insertList,
        method: $.RequestMethod.POST,
        header: $.jsonHeader,
    });
}

const page = function (pagingParam) {
    return $.request({
        url: '/user-for-form/page',
        data: pagingParam,
        method: $.RequestMethod.POST,
        header: $.jsonHeader,
    })
}

const pageForUploadImg = function (pagingParam) {
    return $.request({
        url: '/user-for-form/pageForUploadImg',
        data: pagingParam,
        method: $.RequestMethod.POST,
        header: $.jsonHeader,
    })
}

const pageForSign = function (pagingParam) {
    return $.request({
        url: '/user-for-form/pageForSign',
        data: pagingParam,
        method: $.RequestMethod.POST,
        header: $.jsonHeader,
    })
}


function pageDetailUserForm(pagingParam) {
    return $.request({
        url: '/user-for-form/pageDetailUserForm',
        data: pagingParam,
        method: $.RequestMethod.POST,
        header: $.jsonHeader,
    })
}

function loadListByCondition(condition) {
    return $.request({
        url: '/user-for-form/loadListByCondition',
        data: condition,
        method: $.RequestMethod.POST,
        header: $.jsonHeader,
    })
}

function loadChoiceSelectListByCondition(condition) {
    return $.request({
        url: '/user-for-form/loadChoiceSelectListByCondition',
        data: condition,
        method: $.RequestMethod.POST,
        header: $.jsonHeader,
    })
}

function countByCondition(condition) {
    return $.request({
        url: '/user-for-form/countByCondition',
        data: condition,
        method: $.RequestMethod.POST,
        header: $.jsonHeader,
    })
}




module.exports = {
    page: page,
    insertByList: insertByList,
    pageForUploadImg: pageForUploadImg,
    pageForSign: pageForSign,
    pageDetailUserForm: pageDetailUserForm,
    loadListByCondition: loadListByCondition,
    loadChoiceSelectListByCondition: loadChoiceSelectListByCondition,
    countByCondition: countByCondition,
}

File: back-end/rongshetong-backend/server-bc/src/main/java/com/shopping/wx/service/community_recruitment/UserForFormService.java
package com.shopping.wx.service.community_recruitment;

import com.shopping.wx.model.UserForForm;
import com.shopping.wx.pojo.vo.basic.PagingParam;
import com.shopping.wx.pojo.vo.user_for_form.ChoiceSelectListVo;
import com.shopping.wx.pojo.vo.user_for_form.DetailUserFormVo;
import com.shopping.wx.pojo.vo.user_for_form.SignStatisticVo;
import com.shopping.wx.pojo.vo.user_for_form.UserForFormSearchCondition;
import com.shopping.wx.service.basic.CrudService;

import java.util.List;

/**
 * @InterfaceName UserForFormService
 * @Description TODO
 * @Author zyw
 * @Date 2022/6/14
 **/
public interface UserForFormService extends CrudService<UserForForm> {

    /***
     * 批量插入
     *
     * @param listUserForForm
     * @return java.lang.Integer
     */
    Integer insertByList(List<UserForForm> listUserForForm);

    /**
     * 基础分页
     *
     * @param pagingParam
     * @return java.util.List<com.shopping.wx.model.UserForForm>
     */
    List<UserForForm> page(PagingParam<UserForFormSearchCondition> pagingParam);


    List<UserForForm> pageForUploadImg(PagingParam<UserForFormSearchCondition> pagingParam);

    List<SignStatisticVo> pageForSign(PagingParam<UserForFormSearchCondition> pagingParam);

    List<DetailUserFormVo> pageDetailUserForm(PagingParam<UserForFormSearchCondition> pagingParam);

    List<UserForForm> loadListByCondition(UserForFormSearchCondition condition);

    List<ChoiceSelectListVo> loadChoiceSelectListByCondition(UserForFormSearchCondition condition);


    Integer countByCondition(UserForFormSearchCondition condition);
}
